packages/web-app-external/src/App.vue

```diff
@@ -132,12 +150,54 @@ export default defineComponent({
       }
     }).restartable()
 
+    const determineOpenAsPreview = (appName: string) => {
+      const openAsPreview = configurationManager.options.editors.openAsPreview
+      return (
+        openAsPreview === true || (Array.isArray(openAsPreview) && openAsPreview.includes(appName))
+      )
+    }
+
+    // switch to write mode when edit is clicked
+    const catchClickMicrosoftEdit = (event) => {
+      try {
+        if (JSON.parse(event.data)?.MessageId === 'UI_Edit') {
+          loadAppUrl.perform('write')
+        }
+      } catch (e) {}
+    }
     watch(
-      props.resource,
-      () => {
-        loadAppUrl.perform()
+      applicationName,
+      (newAppName, oldAppName) => {
+        if (determineOpenAsPreview(newAppName) && newAppName !== oldAppName) {
+          window.addEventListener('message', catchClickMicrosoftEdit)
+        } else {
+          window.removeEventListener('message', catchClickMicrosoftEdit)
+        }
+      },
+      {
+        immediate: true
+      }
+    )
+
+    watch(
+      [props.resource],
+      ([newResource], [oldResource]) => {
+        if (isSameResource(newResource, oldResource)) {
+          return
+        }
+
+        debugger
```

Comment:
Maybe not

Reply:
Hehe I fixed that already, maybe I forgot to push